tests/functions.tests

```diff
@@ -51,6 +51,20 @@ fun (x) {x:a} : a
 stderr : @.*
 exit : 1
 
+Infers effect variables
+fun (f, x: Int) { f(x) + 0 }
+stdout : @fun : \(\(Int\) -a-> Int, Int\) -a-> Int
+
+Printer shares type variables
+fun (f, x: Int) { f(x) + 0 }
+stdout : @fun : \(\(Int\) -> Int, Int\) -> Int
+args : --config=tests/functions.tests.shared_effect_vars.config
+
+Parser shares type variables
+fun (f, x: Int) { f(x) + 0 } : ((Int) -> Int, Int) -> Int
+stdout : @fun : \(\(Int\) -> Int, Int\) -> Int
+args : --config=tests/functions.tests.shared_effect_vars.config
+
```

Comment:
Should we have a test case for when the type variables are different, and we should therefore not print `->`?

Reply:
I think the `pickleCont` test (lines 68-71) does that (although I'm not sure if the current behaviour is ideal).  

I don't know if it'd better just to have a standalone definition though, rather than pulling from the prelude?

Reply:
Yes, we have functions in the prelude that exhibit the behavior I'm interested in for the test case. But in the prelude, we don't check that the printer handles such cases correctly.

Reply:
Urg, I misread your comment as "we have such a function in the prelude". What you did is fine.